src/components/solution-control/solution-create-dialog/solution-create-dialog.tsx

```diff
@@ -21,23 +22,37 @@ export const SolutionCreateDialog = ({
   onCreate,
 }: ISolutionCreateDialogProps): ReactElement => {
   const [solutionName, setSolutionName] = useState('')
+  const [isValidationError, setIsValidationError] = useState(false)
 
-  const error = useMemo(
+  const isSameNameError = useMemo(
```

Comment:
That's some unpleasant inconsistency here. If you already have a `useEffect` that watches `solutionName` you can use it for checking if solutions contain the current solution name. And create a useState for it too.
Could have been:
```
const [isValidationError, setIsValidationError] = useState(false)
const [isSameName, setIsSameName] = useState(false)
```